pydicom/dataset.py

```diff
@@ -35,6 +35,7 @@
 
 if TYPE_CHECKING:
     try:
+        import numpy
```

Comment:
Why this is needed? Is it for type-checking?

Reply:
It's for sphinx. I noticed that having "np.ndarray" as the type meant it wouldn't resolve properly in the API docs to the numpy ndarray doc page, but "numpy.ndarray" would. mypy was happy with either.